### ISSUE TITLE
WARNING ldap - No group found for: xxx

### ISSUE THREAD

**Issue by cfibo** [adobe-apiplatform/user-sync.py#360]
Hi

I have an issue I can't understand what is the root cause.

In the Enterprise AD Directory I have a group where I put the users that shall get a Adobe license.

I want to sync only those members with this command
user-sync --users mapped --process-groups --adobe-only-user-action exclude

It works to sync "--users all" but they don't get mapped.
Here is what I have in the user-sync-config:
Groups: 
      - directory_group: "Available Adobe_AcrobatDC"
      adobe_groups:
        - "Acrobat Pro DC User Sync"

connector-ldap has these lines:
all_users_filter: "(&(objectClass=user)(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2)))"
group_filter_format: "(&(|(objectCategory=group)(objectClass=groupOfNames)(objectClass=posixGroup))(cn={group}))"
group_member_filter_format: "(memberOf:1.2.840.113556.1.4.1941:={group_dn})"

The problem is "WARNING ldap - No group found for: Available Adobe_AcrobatDC" and it doesn't map the user to the Adobe Group.
It doesn't find the groups in the AD...
Any hints please?

2018-06-01 10:36:04 8596 INFO config - Using main config file: user-sync-config.yml
2018-06-01 10:36:04 8596 INFO main - ========== Start Run (User Sync version: 2.2.2) ============
2018-06-01 10:36:04 8596 INFO main - ------- Invocation parameters -------
2018-06-01 10:36:04 8596 INFO main - . -t --users mapped --user-filter test.testsson@example.com --process-groups --adobe-only-user-action exclude
2018-06-01 10:36:04 8596 INFO main - -------------------------------------
2018-06-01 10:36:05 8596 INFO processor - ---------- Start Load from Directory -----------------------
2018-06-01 10:36:05 8596 WARNING ldap - No group found for: Available Adobe_AcrobatDC
2018-06-01 10:36:05 8596 INFO processor - ---------- End Load from Directory (Total time: 0:00:00) ---
2018-06-01 10:36:05 8596 INFO processor - ---------- Start Sync with UMAPI ---------------------------
2018-06-01 10:36:06 8596 INFO processor - ---------- End Sync with UMAPI (Total time: 0:00:00) -------
2018-06-01 10:36:06 8596 INFO processor - ---------------------------- Action Summary (TEST MODE) ----------------------------

**Comment by james-d-elliott** [comment]
1. If your LDAP source is Active Directory your group filter format is wrong. If you're using OpenLDAP it's correct. If you're using Active Directory it should be the following:
group_filter_format: "(&(objectCategory=group)(cn={group}))"

2. Do you have username/password/host/base_dn properties set?

3. Check the LDAP query returns values for your LDAP install. For Active Directory you can use Powershell  and use the value of group_member_filter_format if you replace {group} with the value from your group config. See detailed example below:

user-sync-config.yml:
groups:
- directory_group: "Available Adobe_AcrobatDC"
adobe_groups:
- "Acrobat Pro DC User Sync"

connector-ldap.yml:
group_filter_format: "(&(objectCategory=group)(cn={group}))"


Run the following command in powershell:
Get-ADObject -LDAPFilter "(&(objectCategory=group)(cn=Available Adobe_AcrobatDC))"

**Comment by cfibo** [comment]
Found the problem with your help! Thanks so much James - you're a legend!

I did the PowerShell command and that all worked fine so then I knew LDAP returned the correct values.

Then I changed to group_filter_format: "(&(objectCategory=group)(cn={group}))"
but that didn't make any difference 

BUT - the issue was in the domain base definition in "connector-ldap.yml"

Here is what I had:
host: ldap://internal.ourdomain.com
base_dn: OU=Organisation,DC=internal,DC=ourdomain,DC=com

Then I removed the "OU=Organisation" and that made it work 
base_dn: DC=internal,DC=ourdomain,DC=com

Thanks once again! Great help!
